src/rebar_prv_shell.erl

```diff
@@ -328,7 +328,8 @@ reread_config(State) ->
         ConfigList ->
             try
                 [application:set_env(Application, Key, Val)
-                  || {Application, Items} <- ConfigList,
+                  || Config <- ConfigList,
+                     {Application, Items} <- Config,
```

Comment:
From how the code looked this would have never worked, would it? I can easily add this functionality to the PR, though.